tests/functional/test_endpoints_v2.py

```diff
@@ -0,0 +1,372 @@
+                        'endpoint': {
+                            'url': {
+                                'fn': 'getAttr',
+                                'argv': [
+                                    {'ref': 'parsedEndpoint'},
+                                    'authority',
+                                ],
+                            },
+                            'properties': {
+                                "authSchemes": [
+                                    {
+                                        "name": "v4",
+                                        "signingName": "{bucketArn#service}",
+                                        "signingScope": "{Region}",
+                                        "uriEscapePath": False,
+                                    },
+                                ]
+                            },
+                            'headers': {
+                                'x-amz-region-set': [{'ref': 'Region'}],
+                            },
+                        },
+                    },
+                    {
+                        'type': 'error',
+                        'conditions': [
+                            {
+                                'fn': 'not',
+                                'argv': [
+                                    {
+                                        'fn': 'isSet',
+                                        'argv': [
+                                            {
+                                                'fn': 'getAttr',
+                                                'argv': [
+                                                    {'ref': 'bucketArn'},
+                                                    'resourceId[2]',
+                                                ],
+                                            },
+                                        ],
+                                    }
+                                ],
+                            }
+                        ],
+                        'error': {
+                            'fn': 'getAttr',
+                            'argv': [
+                                {'ref': 'bucketArn'},
+                                'service',
+                            ],
+                        },
+                    },
+                    {
+                        'type': 'error',
+                        'conditions': [],
+                        'error': {'ref': 'defaultErrorMessage'},
+                    },
+                ],
+            }
+        ],
+    }
+
+
+@pytest.mark.parametrize(
+    "input_parameters,expected_endpoint",
+    [
+        (
+            {
+                'Region': 'us-east-1',
+                'Bucket': 'arn:aws:s3:us-west-2:123456789012:accesspoint:myendpoint',
+                'Endpoint': 'https://mygreatendpoint.com',
+            },
+            RulesetEndpoint(
+                url='https%3A%2F%2Fmygreatendpoint.com',
```

Comment:
because `uri_encode` looks at an entire value, not just a path. I know it doesn't particularly make sense, but the purpose of these additional handwritten tests was to cover any paths the generated tests did not. I can update to something that would be more valid.

Reply:
unfortunately there are no test cases that use this standard library function yet for some reason

Reply:
yeah i checked with him. there's an example in the vanilla tests. i'd just been using the actual functional tests